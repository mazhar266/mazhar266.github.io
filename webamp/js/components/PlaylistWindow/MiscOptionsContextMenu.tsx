import React from "react";
import { connect } from "react-redux";
import { Node } from "../ContextMenu";
import ContextMenuTarget from "../ContextMenuTarget";
import { Dispatch } from "../../types";
import { downloadHtmlPlaylist } from "../../actionCreators";

interface DispatchProps {
  downloadHtmlPlaylist: () => void;
}

const MiscOptionsContextMenu = (props: DispatchProps) => (
  <ContextMenuTarget
    style={{ width: "100%", height: "100%" }}
    top
    handle={<div />}
  >
    <Node onClick={props.downloadHtmlPlaylist} label="Generate HTML playlist" />
  </ContextMenuTarget>
);

const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => {
  return {
    downloadHtmlPlaylist: () => dispatch(downloadHtmlPlaylist()),
  };
};
export const ConnectedMiscOptionsContextMenu = connect(
  null,
  mapDispatchToProps
)(MiscOptionsContextMenu);
