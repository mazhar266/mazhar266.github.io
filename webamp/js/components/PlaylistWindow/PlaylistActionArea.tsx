import React from "react";
import { connect } from "react-redux";

import {
  play,
  pause,
  stop,
  next,
  previous,
  openMediaFileDialog,
} from "../../actionCreators";

import MiniTime from "../MiniTime";
import RunningTimeDisplay from "./RunningTimeDisplay";
import { Dispatch } from "../../types";

interface Props {
  previous: () => void;
  play: () => void;
  pause: () => void;
  stop: () => void;
  next: () => void;
  openMediaFileDialog: () => void;
}

const PlaylistWindow = (props: Props) => (
  <React.Fragment>
    <RunningTimeDisplay />
    <div className="playlist-action-buttons">
      <div className="playlist-previous-button" onClick={props.previous} />
      <div className="playlist-play-button" onClick={props.play} />
      <div className="playlist-pause-button" onClick={props.pause} />
      <div className="playlist-stop-button" onClick={props.stop} />
      <div className="playlist-next-button" onClick={props.next} />
      <div
        className="playlist-eject-button"
        onClick={props.openMediaFileDialog}
      />
    </div>
    <MiniTime />
  </React.Fragment>
);

const mapDispatchToProps = (dispatch: Dispatch): Props => {
  return {
    play: () => dispatch(play()),
    pause: () => dispatch(pause()),
    stop: () => dispatch(stop()),
    openMediaFileDialog: () => dispatch(openMediaFileDialog()),
    next: () => dispatch(next()),
    previous: () => dispatch(previous()),
  };
};

export default connect(
  null,
  mapDispatchToProps
)(PlaylistWindow);
