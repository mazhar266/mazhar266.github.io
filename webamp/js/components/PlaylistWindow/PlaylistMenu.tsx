import React from "react";
import classnames from "classnames";

import PlaylistMenuEnry from "./PlaylistMenuEntry";

interface Props {
  id: string;
}

interface State {
  selected: boolean;
}

export default class PlaylistMenu extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { selected: false };
  }

  _handleClick = (e: React.MouseEvent<HTMLDivElement>) => {
    const { target } = e;
    if (!(target instanceof Element)) {
      // TypeScript doesn't realize this will always be true
      return;
    }
    const { selected } = this.state;
    if (selected) {
      this.setState({ selected: false });
      return;
    }

    const handleClickOut = (ee: MouseEvent) => {
      const clickOutTarget = ee.target;
      if (!(clickOutTarget instanceof Element)) {
        // TypeScript doesn't realize this will always be true
        return;
      }
      // If the click is _not_ inside the menu.
      if (!target.contains(clickOutTarget)) {
        // If we've clicked on a Context Menu spawed inside this menu, it will
        // register as an external click. However, hiding the menu will remove
        // the Context Menu from the DOM. Therefore, we wait until the next
        // event loop to actually hide ourselves.
        setTimeout(() => {
          // Close the menu
          this.setState({ selected: false });
        }, 0);
        window.document.removeEventListener("click", handleClickOut, {
          capture: true,
        });
      }
    };
    window.document.addEventListener("click", handleClickOut, {
      capture: true,
    });

    this.setState({ selected: true });
  };

  render() {
    return (
      <div
        id={this.props.id}
        className={classnames("playlist-menu", {
          selected: this.state.selected,
        })}
        onClick={this._handleClick}
      >
        <div className="bar" />
        {this.state.selected && (
          <ul>
            {React.Children.map(this.props.children, (child, i) => (
              <PlaylistMenuEnry key={i}>{child}</PlaylistMenuEnry>
            ))}
          </ul>
        )}
      </div>
    );
  }
}
