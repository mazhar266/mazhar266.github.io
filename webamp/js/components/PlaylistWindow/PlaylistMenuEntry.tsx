import React from "react";
import classnames from "classnames";

interface Props {}

interface State {
  hover: boolean;
}
let cursorX: number;
let cursorY: number;
window.document.addEventListener("mousemove", e => {
  cursorX = e.pageX;
  cursorY = e.pageY;
});

// We implement hover ourselves, because we hate ourselves and https://stackoverflow.com/a/13259049/1263117
export default class PlaylistMenuEntry extends React.Component<Props, State> {
  node?: HTMLLIElement | null;
  constructor(props: Props) {
    super(props);
    this.state = { hover: false };
  }

  componentDidMount() {
    if (this.node == null) {
      return;
    }
    const domRect = this.node.getBoundingClientRect();
    this.setState({
      hover:
        cursorX >= domRect.left &&
        cursorX <= domRect.right &&
        cursorY >= domRect.top &&
        cursorY <= domRect.bottom,
    });
  }

  render() {
    return (
      <li
        ref={node => (this.node = node)}
        onMouseEnter={() => this.setState({ hover: true })}
        onMouseLeave={() => this.setState({ hover: false })}
        className={classnames({ hover: this.state.hover })}
      >
        {this.props.children}
      </li>
    );
  }
}
