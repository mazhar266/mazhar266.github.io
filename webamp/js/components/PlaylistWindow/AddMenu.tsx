import React from "react";
import { connect } from "react-redux";
import { LOAD_STYLE } from "../../constants";
import { getTrackCount } from "../../selectors";
import { addTracksFromReferences } from "../../actionCreators";
import { promptForFileReferences } from "../../fileUtils";
import PlaylistMenu from "./PlaylistMenu";
import { AppState, Dispatch } from "../../types";

interface StateProps {
  nextIndex: number;
}

interface DispatchProps {
  addFilesAtIndex(i: number): void;
  addDirAtIndex(i: number): void;
}
const el = document.createElement("input");
el.type = "file";
// @ts-ingore
const DIR_SUPPORT =
  // @ts-ignore
  typeof el.webkitdirectory !== "undefined" ||
  // @ts-ignore
  typeof el.mozdirectory !== "undefined" ||
  // @ts-ignore
  typeof el.directory !== "undefined";

/* eslint-disable no-alert */

const AddMenu = ({
  nextIndex,
  addFilesAtIndex,
  addDirAtIndex,
}: StateProps & DispatchProps) => (
  <PlaylistMenu id="playlist-add-menu">
    <div className="add-url" onClick={() => alert("Not supported in Webamp")} />
    <div className="add-dir" onClick={() => addDirAtIndex(nextIndex)} />
    <div className="add-file" onClick={() => addFilesAtIndex(nextIndex)} />
  </PlaylistMenu>
);

const mapStateToProps = (state: AppState): StateProps => ({
  nextIndex: getTrackCount(state),
});

const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => ({
  addFilesAtIndex: async nextIndex => {
    const fileReferences = await promptForFileReferences();
    dispatch(
      addTracksFromReferences(fileReferences, LOAD_STYLE.NONE, nextIndex)
    );
  },
  addDirAtIndex: async nextIndex => {
    if (!DIR_SUPPORT) {
      alert("Not supported in your browser");
      return;
    }
    const fileReferences = await promptForFileReferences({ directory: true });
    dispatch(
      addTracksFromReferences(fileReferences, LOAD_STYLE.NONE, nextIndex)
    );
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AddMenu);
