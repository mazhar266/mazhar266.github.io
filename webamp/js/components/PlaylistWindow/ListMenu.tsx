import React from "react";
import { connect } from "react-redux";
import { removeAllTracks } from "../../actionCreators";
import PlaylistMenu from "./PlaylistMenu";

/* eslint-disable no-alert */

interface DispatchProps {
  removeAllTracks: () => void;
}

const ListMenu = (props: DispatchProps) => (
  <PlaylistMenu id="playlist-list-menu">
    <div className="new-list" onClick={props.removeAllTracks} />
    <div
      className="save-list"
      onClick={() => alert("Not supported in Webamp")}
    />
    <div
      className="load-list"
      onClick={() => alert("Not supported in Webamp")}
    />
  </PlaylistMenu>
);

const mapDispatchToProps = (): DispatchProps => {
  return { removeAllTracks };
};
export default connect(
  null,
  mapDispatchToProps
)(ListMenu);
