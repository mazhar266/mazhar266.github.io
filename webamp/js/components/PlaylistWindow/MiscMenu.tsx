import React from "react";

import PlaylistMenu from "./PlaylistMenu";
import SortContextMenu from "./SortContextMenu";
import { ConnectedMiscOptionsContextMenu } from "./MiscOptionsContextMenu";

const MiscMenu = () => (
  <PlaylistMenu id="playlist-misc-menu">
    <div className="sort-list" onClick={e => e.stopPropagation()}>
      <SortContextMenu />
    </div>
    <div
      className="file-info"
      onClick={() => alert("Not supported in Webamp")}
    />

    <div className="misc-options" onClick={e => e.stopPropagation()}>
      <ConnectedMiscOptionsContextMenu />
    </div>
  </PlaylistMenu>
);

export default MiscMenu;
