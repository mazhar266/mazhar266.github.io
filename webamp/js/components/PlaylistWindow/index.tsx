import React from "react";
import { connect } from "react-redux";
import classnames from "classnames";

import { WINDOWS, TRACK_HEIGHT, LOAD_STYLE } from "../../constants";
import {
  scrollUpFourTracks,
  scrollDownFourTracks,
  togglePlaylistShadeMode,
  scrollVolume,
  closeWindow,
  loadMedia,
} from "../../actionCreators";
import * as Selectors from "../../selectors";

import { clamp } from "../../utils";
import DropTarget from "../DropTarget";
import Visualizer from "../Visualizer";
import PlaylistShade from "./PlaylistShade";
import AddMenu from "./AddMenu";
import RemoveMenu from "./RemoveMenu";
import SelectionMenu from "./SelectionMenu";
import MiscMenu from "./MiscMenu";
import ListMenu from "./ListMenu";
import PlaylistResizeTarget from "./PlaylistResizeTarget";
import PlaylistActionArea from "./PlaylistActionArea";
import TrackList from "./TrackList";
import ScrollBar from "./ScrollBar";

import "../../../css/playlist-window.css";
import { AppState, PlaylistStyle, Dispatch } from "../../types";
import FocusTarget from "../FocusTarget";

interface StateProps {
  offset: number;
  maxTrackIndex: number;
  playlistWindowPixelSize: { width: number; height: number };
  showVisualizer: boolean;
  activateVisualizer: boolean;
  selected: boolean;
  skinPlaylistStyle: PlaylistStyle;
  playlistSize: [number, number];
  playlistShade: boolean;
  duration: number | null;
}

interface DispatchProps {
  close(): void;
  toggleShade(): void;
  scrollUpFourTracks(): void;
  scrollDownFourTracks(): void;
  loadMedia(e: React.DragEvent<HTMLDivElement>, startIndex: number): void;
  scrollVolume(e: React.WheelEvent<HTMLDivElement>): void;
}

interface OwnProps {
  analyser: AnalyserNode;
}

type Props = StateProps & DispatchProps & OwnProps;

class PlaylistWindow extends React.Component<Props> {
  _handleDrop = (
    e: React.DragEvent<HTMLDivElement>,
    targetCoords: { x: number; y: number }
  ) => {
    const top = e.clientY - targetCoords.y;
    const atIndex = clamp(
      this.props.offset + Math.round((top - 23) / TRACK_HEIGHT),
      0,
      this.props.maxTrackIndex + 1
    );
    this.props.loadMedia(e, atIndex);
  };

  render() {
    const {
      skinPlaylistStyle,
      selected,
      playlistSize,
      playlistWindowPixelSize,
      playlistShade,
      close,
      toggleShade,
      analyser,
      showVisualizer,
      activateVisualizer,
    } = this.props;
    if (playlistShade) {
      return <PlaylistShade />;
    }

    const style = {
      color: skinPlaylistStyle.normal,
      backgroundColor: skinPlaylistStyle.normalbg,
      fontFamily: `${skinPlaylistStyle.font}, Arial, sans-serif`,
      height: `${playlistWindowPixelSize.height}px`,
      width: `${playlistWindowPixelSize.width}px`,
    };

    const classes = classnames("window", "draggable", { selected });

    const showSpacers = playlistSize[0] % 2 === 0;

    return (
      <FocusTarget windowId={WINDOWS.PLAYLIST}>
        <DropTarget
          id="playlist-window"
          className={classes}
          style={style}
          handleDrop={this._handleDrop}
          onWheel={this.props.scrollVolume}
        >
          <React.StrictMode>
            <div className="playlist-top draggable" onDoubleClick={toggleShade}>
              <div className="playlist-top-left draggable" />
              {showSpacers && (
                <div className="playlist-top-left-spacer draggable" />
              )}
              <div className="playlist-top-left-fill draggable" />
              <div className="playlist-top-title draggable" />
              {showSpacers && (
                <div className="playlist-top-right-spacer draggable" />
              )}
              <div className="playlist-top-right-fill draggable" />
              <div className="playlist-top-right draggable">
                <div id="playlist-shade-button" onClick={toggleShade} />
                <div id="playlist-close-button" onClick={close} />
              </div>
            </div>
          </React.StrictMode>
          <div className="playlist-middle draggable">
            <React.StrictMode>
              <div className="playlist-middle-left draggable" />
              <div className="playlist-middle-center">
                <TrackList />
              </div>
            </React.StrictMode>
            <div className="playlist-middle-right draggable">
              <ScrollBar />
            </div>
          </div>
          <React.StrictMode>
            <div className="playlist-bottom draggable">
              <div className="playlist-bottom-left draggable">
                <AddMenu />
                <RemoveMenu />
                <SelectionMenu />
                <MiscMenu />
              </div>
              <div className="playlist-bottom-center draggable" />
              <div className="playlist-bottom-right draggable">
                {showVisualizer && (
                  <div className="playlist-visualizer">
                    {activateVisualizer && (
                      <div className="visualizer-wrapper">
                        <Visualizer
                          // @ts-ignore Visualizer is not yet typed
                          analyser={analyser}
                        />
                      </div>
                    )}
                  </div>
                )}
                <PlaylistActionArea />
                <ListMenu />
                <div
                  id="playlist-scroll-up-button"
                  onClick={this.props.scrollUpFourTracks}
                />
                <div
                  id="playlist-scroll-down-button"
                  onClick={this.props.scrollDownFourTracks}
                />
                <PlaylistResizeTarget />
              </div>
            </div>
          </React.StrictMode>
        </DropTarget>
      </FocusTarget>
    );
  }
}

const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => {
  return {
    close: () => dispatch(closeWindow(WINDOWS.PLAYLIST)),
    toggleShade: () => dispatch(togglePlaylistShadeMode()),
    scrollUpFourTracks: () => dispatch(scrollUpFourTracks()),
    scrollDownFourTracks: () => dispatch(scrollDownFourTracks()),
    loadMedia: (e, startIndex) =>
      dispatch(loadMedia(e, LOAD_STYLE.NONE, startIndex)),
    scrollVolume: e => dispatch(scrollVolume(e)),
  };
};

const mapStateToProps = (state: AppState): StateProps => {
  const {
    playlist: { trackOrder },
  } = state;
  const playlistSize = Selectors.getWindowSize(state)(WINDOWS.PLAYLIST);

  return {
    offset: Selectors.getScrollOffset(state),
    maxTrackIndex: trackOrder.length - 1,
    playlistWindowPixelSize: Selectors.getWindowPixelSize(state)(
      WINDOWS.PLAYLIST
    ),
    showVisualizer: playlistSize[0] > 2,
    activateVisualizer: !Selectors.getWindowOpen(state)(WINDOWS.MAIN),
    playlistSize,
    selected: Selectors.getFocusedWindow(state) === WINDOWS.PLAYLIST,
    skinPlaylistStyle: Selectors.getSkinPlaylistStyle(state),
    playlistShade: Boolean(Selectors.getWindowShade(state)(WINDOWS.PLAYLIST)),
    duration: Selectors.getDuration(state),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PlaylistWindow);
