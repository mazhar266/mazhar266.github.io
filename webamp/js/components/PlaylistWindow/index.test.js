import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import getStore from "../../store";

import PlaylistWindow from "./index";
const media = {
  addEventListener: jest.fn(),
  setVolume: jest.fn(),
  setPreamp: jest.fn(),
  setBalance: jest.fn(),
  getAnalyser: () => null,
  on: jest.fn(),
};

describe("PlaylistWindow", () => {
  let store;
  beforeEach(() => {
    store = getStore(media);
  });

  it("renders to snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <PlaylistWindow />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
