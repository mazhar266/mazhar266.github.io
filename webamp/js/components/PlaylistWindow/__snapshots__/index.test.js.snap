// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaylistWindow renders to snapshot 1`] = `
<div
  onFocus={[Function]}
  onMouseDown={[Function]}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
  tabIndex={-1}
>
  <div
    className="window draggable"
    id="playlist-window"
    onDragEnter={[Function]}
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onWheel={[Function]}
    style={
      Object {
        "backgroundColor": "#000000",
        "color": "#00FF00",
        "fontFamily": "Arial, Arial, sans-serif",
        "height": "116px",
        "width": "275px",
      }
    }
  >
    <div
      className="playlist-top draggable"
      onDoubleClick={[Function]}
    >
      <div
        className="playlist-top-left draggable"
      />
      <div
        className="playlist-top-left-spacer draggable"
      />
      <div
        className="playlist-top-left-fill draggable"
      />
      <div
        className="playlist-top-title draggable"
      />
      <div
        className="playlist-top-right-spacer draggable"
      />
      <div
        className="playlist-top-right-fill draggable"
      />
      <div
        className="playlist-top-right draggable"
      >
        <div
          id="playlist-shade-button"
          onClick={[Function]}
        />
        <div
          id="playlist-close-button"
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="playlist-middle draggable"
    >
      <div
        className="playlist-middle-left draggable"
      />
      <div
        className="playlist-middle-center"
      >
        <div
          className="playlist-tracks"
          onClick={[Function]}
          onWheel={[Function]}
          style={
            Object {
              "height": "100%",
            }
          }
        >
          <div
            className="playlist-track-titles"
          />
          <div
            className="playlist-track-durations"
          />
        </div>
      </div>
      <div
        className="playlist-middle-right draggable"
      >
        <div
          className="rc-slider rc-slider-disabled rc-slider-vertical playlist-scrollbar"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="rc-slider-rail"
            style={Object {}}
          />
          <div
            className="rc-slider-track"
            style={
              Object {
                "bottom": "0%",
                "height": "0%",
              }
            }
          />
          <div
            className="rc-slider-step"
          />
          <div
            className="playlist-scrollbar-handle"
          />
          <div
            className="rc-slider-mark"
          />
        </div>
      </div>
    </div>
    <div
      className="playlist-bottom draggable"
    >
      <div
        className="playlist-bottom-left draggable"
      >
        <div
          className="playlist-menu"
          id="playlist-add-menu"
          onClick={[Function]}
        >
          <div
            className="bar"
          />
        </div>
        <div
          className="playlist-menu"
          id="playlist-remove-menu"
          onClick={[Function]}
        >
          <div
            className="bar"
          />
        </div>
        <div
          className="playlist-menu"
          id="playlist-selection-menu"
          onClick={[Function]}
        >
          <div
            className="bar"
          />
        </div>
        <div
          className="playlist-menu"
          id="playlist-misc-menu"
          onClick={[Function]}
        >
          <div
            className="bar"
          />
        </div>
      </div>
      <div
        className="playlist-bottom-center draggable"
      />
      <div
        className="playlist-bottom-right draggable"
      >
        <div
          className="playlist-running-time-display draggable"
        >
          <div>
            <span
              className=" character character-48"
            >
              0
            </span>
            <span
              className=" character character-58"
            >
              :
            </span>
            <span
              className=" character character-48"
            >
              0
            </span>
            <span
              className=" character character-48"
            >
              0
            </span>
            <span
              className=" character character-47"
            >
              /
            </span>
            <span
              className=" character character-48"
            >
              0
            </span>
            <span
              className=" character character-58"
            >
              :
            </span>
            <span
              className=" character character-48"
            >
              0
            </span>
            <span
              className=" character character-48"
            >
              0
            </span>
            <span
              className=" character character-32"
            >
               
            </span>
            <span
              className=" character character-32"
            >
               
            </span>
            <span
              className=" character character-32"
            >
               
            </span>
            <span
              className=" character character-32"
            >
               
            </span>
            <span
              className=" character character-32"
            >
               
            </span>
            <span
              className=" character character-32"
            >
               
            </span>
            <span
              className=" character character-32"
            >
               
            </span>
            <span
              className=" character character-32"
            >
               
            </span>
            <span
              className=" character character-32"
            >
               
            </span>
          </div>
        </div>
        <div
          className="playlist-action-buttons"
        >
          <div
            className="playlist-previous-button"
            onClick={[Function]}
          />
          <div
            className="playlist-play-button"
            onClick={[Function]}
          />
          <div
            className="playlist-pause-button"
            onClick={[Function]}
          />
          <div
            className="playlist-stop-button"
            onClick={[Function]}
          />
          <div
            className="playlist-next-button"
            onClick={[Function]}
          />
          <div
            className="playlist-eject-button"
            onClick={[Function]}
          />
        </div>
        <div
          className="mini-time countdown"
          onClick={[Function]}
        >
          <span
            className="background-character character character-32"
            style={
              Object {
                "left": 1,
              }
            }
          >
             
          </span>
          <span
            className="background-character character character-32"
            style={
              Object {
                "left": 7,
              }
            }
          >
             
          </span>
          <span
            className="background-character character character-32"
            style={
              Object {
                "left": 12,
              }
            }
          >
             
          </span>
          <span
            className="background-character character character-32"
            style={
              Object {
                "left": 20,
              }
            }
          >
             
          </span>
          <span
            className="background-character character character-32"
            style={
              Object {
                "left": 25,
              }
            }
          >
             
          </span>
          <span
            className=" character character-32"
            style={
              Object {
                "left": 1,
              }
            }
          >
             
          </span>
          <span
            className=" character character-32"
            style={
              Object {
                "left": 7,
              }
            }
          >
             
          </span>
          <span
            className=" character character-32"
            style={
              Object {
                "left": 12,
              }
            }
          >
             
          </span>
          <span
            className=" character character-32"
            style={
              Object {
                "left": 20,
              }
            }
          >
             
          </span>
          <span
            className=" character character-32"
            style={
              Object {
                "left": 25,
              }
            }
          >
             
          </span>
        </div>
        <div
          className="playlist-menu"
          id="playlist-list-menu"
          onClick={[Function]}
        >
          <div
            className="bar"
          />
        </div>
        <div
          id="playlist-scroll-up-button"
          onClick={[Function]}
        />
        <div
          id="playlist-scroll-down-button"
          onClick={[Function]}
        />
        <div
          id="playlist-resize-target"
          onMouseDown={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
